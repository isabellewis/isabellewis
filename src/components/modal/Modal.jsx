import React, { PureComponent } from 'react';
import {
  ModalWrapper,
  OuterWrapper,
  CloseIcon
} from './Modal.style';

export class Modal extends PureComponent {
	constructor(props) {
		super(props);
	}

	render() {
		return(
			<OuterWrapper className={this.props.open ? 'open' : 'closed'} id="hello">
				<ModalWrapper>
					<CloseIcon className="material-icons" onClick={this.props.toggle}>close</CloseIcon>
					<h2>Hello yes helloooo</h2>
					<p>{this.props.text}</p>
				</ModalWrapper>
			</OuterWrapper>
		);
	}
} 
