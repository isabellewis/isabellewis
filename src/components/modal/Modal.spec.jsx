import React from 'react';
import { shallow } from 'enzyme';
import { Modal } from './Modal';

describe('Modal', () => {
	let component;

	beforeEach(() => {
		component = shallow(<Modal open={true} text='Modal test text' />);
	});

	it('should match the snapshot when open', () => {
		expect(component).toMatchSnapshot();
	});

	it('should match the snapshot when closed', () => {
		component = shallow(<Modal open={false} text='Modal test text' />);
		expect(component).toMatchSnapshot();
	});
});