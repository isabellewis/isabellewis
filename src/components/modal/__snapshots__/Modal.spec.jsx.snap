// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match the snapshot when closed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    open={false}
    text="Modal test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.i
          className="material-icons"
          onClick={undefined}
        >
          close
        </styled.i>
        <h2>
          Hello yes helloooo
        </h2>
        <p>
          Modal test text
        </p>
      </styled.div>,
      "className": "closed",
      "id": "hello",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.i
            className="material-icons"
            onClick={undefined}
          >
            close
          </styled.i>,
          <h2>
            Hello yes helloooo
          </h2>,
          <p>
            Modal test text
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "close",
            "className": "material-icons",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello yes helloooo",
          },
          "ref": null,
          "rendered": "Hello yes helloooo",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Modal test text",
          },
          "ref": null,
          "rendered": "Modal test text",
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.i
            className="material-icons"
            onClick={undefined}
          >
            close
          </styled.i>
          <h2>
            Hello yes helloooo
          </h2>
          <p>
            Modal test text
          </p>
        </styled.div>,
        "className": "closed",
        "id": "hello",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.i
              className="material-icons"
              onClick={undefined}
            >
              close
            </styled.i>,
            <h2>
              Hello yes helloooo
            </h2>,
            <p>
              Modal test text
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "close",
              "className": "material-icons",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "close",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello yes helloooo",
            },
            "ref": null,
            "rendered": "Hello yes helloooo",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Modal test text",
            },
            "ref": null,
            "rendered": "Modal test text",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Modal should match the snapshot when open 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    open={true}
    text="Modal test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.i
          className="material-icons"
          onClick={undefined}
        >
          close
        </styled.i>
        <h2>
          Hello yes helloooo
        </h2>
        <p>
          Modal test text
        </p>
      </styled.div>,
      "className": "open",
      "id": "hello",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.i
            className="material-icons"
            onClick={undefined}
          >
            close
          </styled.i>,
          <h2>
            Hello yes helloooo
          </h2>,
          <p>
            Modal test text
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "close",
            "className": "material-icons",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello yes helloooo",
          },
          "ref": null,
          "rendered": "Hello yes helloooo",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Modal test text",
          },
          "ref": null,
          "rendered": "Modal test text",
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.i
            className="material-icons"
            onClick={undefined}
          >
            close
          </styled.i>
          <h2>
            Hello yes helloooo
          </h2>
          <p>
            Modal test text
          </p>
        </styled.div>,
        "className": "open",
        "id": "hello",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.i
              className="material-icons"
              onClick={undefined}
            >
              close
            </styled.i>,
            <h2>
              Hello yes helloooo
            </h2>,
            <p>
              Modal test text
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "close",
              "className": "material-icons",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "close",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello yes helloooo",
            },
            "ref": null,
            "rendered": "Hello yes helloooo",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Modal test text",
            },
            "ref": null,
            "rendered": "Modal test text",
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
